components/base/kinematicbase/execution.go

```diff
@@ -505,24 +504,17 @@ func (ptgk *ptgBaseKinematics) getCorrectionSolution(ctx context.Context, goals
 		}
 		// Attempt to use our course correction solver to solve for a new set of trajectories which will get us from our current position
 		// to our goal point along our original trajectory.
-		err := ptgk.ptgs[ptgk.courseCorrectionIdx].Solve(
+		solution, err := ptgk.ptgs[ptgk.courseCorrectionIdx].Solve(
 			ctx,
-			solutionChan,
 			seed,
 			solveMetric,
-			0,
 		)
 		if err != nil {
 			return courseCorrectionGoal{}, err
 		}
-		var solution *ik.Solution
-		select {
-		case solution = <-solutionChan:
-		default:
-		}
 		ptgk.logger.Debug("solution ", solution)
 		if solution.Score < courseCorrectionMaxScore {
-			goal.Solution = solution.Configuration
+			goal.Solution = referenceframe.FloatsToInputs(solution.Configuration)
```

Comment:
No; we are moving away from `Inputs` only in `ik`, because `Inputs` are specifically tied to `Frames`, and we are removing the requirement that `ik` operate on a `Frame`.

Reply:
Inputs are specifically meant to be used with Frames; like, specific Frames.

Without an associated Frame around, an Input is just a bonus layer of abstraction around a float.

Reply:
Then I'm not sure what the point of Inputs even is at this point.  They're useful for converting from JointPositions but at this point thats all they're doing

Reply:
They've always been a really thin wrapper around floats and I think that if we're going to call them bonus layers of abstraction and need to convert between floats in between our own packages, then we need to rethink why we have them in the first place.

Also, as much as I want this package to be able to stand alone, we should be building this package assuming that we are the only clients of it and I think that tying it to Inputs means we have less to translate between our layers of the stack.

Reply:
ok I'm starting to understand your thought process.

Reply:
I would consider type aliases such as Inputs generally a useful/beneficial tool even in limited circumstances. Consider the utility of, for example, `time.Duration`